File: backend/masz/Models/Views/QuickSearchEntry.cs
using MASZ.Enums;

namespace MASZ.Models
{
    public interface IQuickSearchEntry
    {
        DateTime CreatedAt { get; set; }
    }

    public class QuickSearchEntry<T> : IQuickSearchEntry
    {
        public T Entry { get; set; }
        public DateTime CreatedAt { get; set; }
        public QuickSearchEntryType QuickSearchEntryType { get; set; }
    }
}

File: nginx/MASZ/src/app/models/QuickSearchEntry.ts
import { AutoModerationEventTableEntry } from "./AutoModerationEventTableEntry";
import { IModCaseTableEntry } from "./IModCaseTableEntry";

export interface QuickSearchEntry {
    entry: IModCaseTableEntry|AutoModerationEventTableEntry;
    createdAt: Date;
    quickSearchEntryType: number;
}